astropy/cosmology/tests/test_core.py

```diff
@@ -0,0 +1,555 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""Testing :mod:`astropy.cosmology.core`."""
+
+##############################################################################
+# IMPORTS
+
+import inspect
+import sys
+import weakref
+from io import StringIO, TextIOWrapper
+from types import MappingProxyType
+
+import pytest
+
+import numpy as np
+
+import astropy.units as u
+from astropy.cosmology import Cosmology, FlatLambdaCDM, Planck18, core
+from astropy.cosmology.core import _COSMOLOGY_CLASSES, Parameter
+
+##############################################################################
+# TESTS
+##############################################################################
+
+
+class TestCosmologyInfo:
+    """Test `astropy.cosmology.core.CosmologyInfo`."""
+
+    @pytest.fixture(params=[None, "", sys.stdout, StringIO])
+    def out(self, request):
+        out = request.param
+        if out is StringIO:
+            out = out()
+
+        return out
+
+    @pytest.fixture
+    def cosmo(self):
+        return Planck18
+
+    @pytest.fixture
+    def cosmo_cls(self):
+        return Planck18.__class__
+
+    @pytest.fixture
+    def info(self, cosmo):
+        # yes, it's a descriptor
+        return cosmo.info
+
+    # ==============================================================
+
+    def test_has_expected_attributes(self, info):
+        """Test ``CosmologyInfo`` has all the expected attributes."""
+        assert info._stats == []
```

Comment:
I'd generally not test private attributes, but rather (try to) test their effects.